implementation.md

```diff
@@ -42,7 +42,7 @@ RFC 3339 is a profile of ISO 8601, adding these constraints to the allowed value
 A simple regex for an RFC 3339 datetime is:
 
 ```
-^(\d\d\d\d)\-(\d\d)\-(\d\d)(T|t)(\d\d):(\d\d):(\d\d)(\.\d+)?(Z|([-+])(\d\d):(\d\d))$
+^(\d\d\d\d)\-(\d\d)\-(\d\d)(T|t)(\d\d):(\d\d):(\d\d)([,.]\d+)?(Z|([-+])(\d\d):(\d\d))$
```

Comment:
AFAIK the dot needs to be escaped, otherwise it matches all characters.

Reply:
Inside `[]`, `.` is a literal and not a metacharacter like it is normally.

Reply:
Nice, still learning new stuff about regexes.

Reply:
I did this notebook a couple of months ago to help out my understanding -- I can never remember to rules. https://github.com/philvarner/old-possum/blob/main/scala/RegexExamples.ipynb It's scala, but the strings are standard regex. 

One thing I found out because of a bug is that `-` in `[]` is a range character if it's between two other characters, but a literal if it's the first character or if it's escaped. So, I always put it as the first character if I'm matching on it, e.g., `[.-_]` matches the ascii range between `.` and `_` but `[-._]` matches those three literal chars. That's why I wrote -+ in the timezone offset above instead of +-

Reply:
Yeah, I knew about the dash issue. Thanks for the link.